iot-hub/Samples/device/DeviceReconnectionSample/ColorConsoleLogger.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    public class ColorConsoleLogger : ILogger
+    {
+        private readonly string _name;
+        private readonly ColorConsoleLoggerConfiguration _config;
+
+        public ColorConsoleLogger(string name, ColorConsoleLoggerConfiguration config)
+        {
+            _name = name;
+            _config = config;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel == _config.LogLevel;
```

Comment:
I assume logging level is a gradually increasing level (e.g. verbose, info, warning, error), in which case, shouldn't this be >= (or <= depending on the numeric order)?

Reply:
Based on the rest of the code I read, it looks like a console logger is added for a specific level. All this just to have different colors per level?

Reply:
Yes, all of this is just to add color to the console logs. On directing the sdk logs to the console output it was getting a bit difficult parsing out the info; and the ConsoleLogger available colors only the log level info, and not the log message. 
I'll add a note to the sample highlighting this.

Reply:
Why not one logger with colors specified per color? Maybe it could have defaults that are overridable via the ctor?

Reply:
Sure, that would definitely be an improvement over the current implementation. I can add that in a subsequent PR, since the focus for this one is the reconnection part of the sample. 
Do you think the logging is too confusing, the way it is right now?

Reply:
It doesn't work the way I've seen most loggers where logLevel has more to do with reducing the logging output during common activities, so it threw me for a loop. I imagine it would be similarly confusing for others for that reason too.

Reply:
I see, I'll see if I can make some quick simplifications, else I'll add the provided ConsoleLogger for now, and add this colored version later.